src/components/MessagesList/MessagesGroup/MessagesSystemGroup.vue

```diff
@@ -223,6 +230,11 @@ export default {
 					if (isLastReadInsideGroup) {
 						groups.at(-1).collapsed = false
 					}
+
+					// Update collapsed state for the last group if it's not defined yet
+					if (JSON.stringify(message) === JSON.stringify(messages.at(-1)) && groups.at(-1).collapsed === undefined) {
+						groups.at(-1).collapsed = groups.at(-1).messages.length > 1
+					}
 				}
 				lastMessage = message
 			}
```

Comment:
Instead of stringifying last messages every, i think it would be more performant to move out all logic to another cycle, there we iterate over compiled group and decide on each one, whether it's supposed to be collapsed
```suggestion
			}
			
			for (const group of groups) {
				// Iterate over all groups and apply collapsed state once (adjust your logic)
				group.collapsed = this.groupIsCollapsed[message.id] ?? <isLastReadInsideGroup || messages.length > 1>
			}
```